BranchSDK/BranchEvent.m

```diff
@@ -367,4 +369,104 @@ - (NSString*_Nonnull) description {
     ];
 }
 
+#pragma mark - IAP Methods
+
+- (void) logEventWithTransaction:(SKPaymentTransaction *)transaction {
+    
+    self.transactionID = transaction.transactionIdentifier;
+    [[BNCEventUtils shared] storeEvent:self];
+
+    NSString *productId = transaction.payment.productIdentifier;
+    SKProductsRequest *productsRequest = [[SKProductsRequest alloc] initWithProductIdentifiers:[NSSet setWithObject:productId]];
+    
+    _request = productsRequest;
+    productsRequest.delegate = self;
+    [productsRequest start];
+}
+
+- (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProductsResponse *)response {
```

Comment:
The docs have a note that we cannot assume which thread this comes in on. I don't think we're doing anything too expensive or blocking, but it might be a good idea to dispatch this anyway.